vignettes/examples/variational_autoencoder.R

```diff
@@ -1 +1,131 @@
+# This script demonstrates how to build a variational autoencoder with Keras.
+# Reference: "Auto-Encoding Variational Bayes" https://arxiv.org/abs/1312.6114
 library(keras)
+K <- keras::backend()
+
+# Parameters --------------------------------------------------------------
+
+batch_size <- 100L
+original_dim <- 784L
+latent_dim <- 2L
+intermediate_dim <- 256L
+epochs <- 50L
+epsilon_std <- 1.0
+
+# Model definition --------------------------------------------------------
+
+x <- layer_input(batch_shape = c(batch_size, original_dim))
+h <- layer_dense(x, intermediate_dim, activation = "relu")
+z_mean <- layer_dense(h, latent_dim)
+z_log_var <- layer_dense(h, latent_dim)
+
+sampling <- function(arg){
+  z_mean <- arg[,0:1]
+  z_log_var <- arg[,2:3]
+  
+  epsilon <- K$random_normal(
+    shape = c(batch_size, latent_dim), 
+    mean=0.,
+    stddev=epsilon_std
+  )
+  
+  z_mean + K$exp(z_log_var/2)*epsilon
+}
+
+# note that "output_shape" isn't necessary with the TensorFlow backend
+z <- layer_concatenate(list(z_mean, z_log_var)) %>% 
```

Comment:
How does `layer_concatenate` works? It´s very similar to passing a `list` to `layer_lambda`. The python equivalent code uses a list in this example:

https://github.com/rstudio/keras/blob/master/vignettes/examples/variational_autoencoder.py#L29-L36